src/common/file_system.cpp

```diff
@@ -703,7 +703,7 @@ void FileHandle::Read(void *buffer, idx_t nr_bytes, idx_t location) {
 	file_system.Read(*this, buffer, UnsafeNumericCast<int64_t>(nr_bytes), location);
 }
 
-void FileHandle::Write(optional_ptr<ClientContext> context, void *buffer, idx_t nr_bytes, idx_t location) {
+void FileHandle::Write(QueryContext &context, void *buffer, idx_t nr_bytes, idx_t location) {
```

Comment:
Since `QueryContext` is just a wrapper for a pointer - can we make this not a reference but just `QueryContext context`? Making it a reference means we can't do `Write(QueryContext(), ...)` and means there are now life-time considerations that are not really necessary